addons/arrays/fnc_prettyFormat.sqf

```diff
@@ -0,0 +1,71 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_prettyFormat
+
+Description:
+    Makes an array easier to read.
+
+Parameters:
+    _array - The array to format <ARRAY>
+    _tab   - The tab char to use (optional, default: tab) <STRING>
+    _depth - Starting indent (optional, default: 0) <NUMBER>
+
+Returns:
+    Formatted string <STRING>
+
+Examples:
+    (begin example)
+        _return = [ [0, 1, ["22", 33, []], 4] ] call CBA_fnc_prettyFormat;
+        // _return ==>
+        // "[
+        //      0,
+        //      1,
+        //      [
+        //          ""22"",
+        //          33,
+        //          []
+        //      ],
+        //      4
+        //  ]"
+        _return = [ [0, 1, ["22", 33, []], 4], ">---" ] call CBA_fnc_prettyFormat;
+        // _return ==>
+        // [
+        // >---0,
+        // >---1,
+        // >---[
+        // >--->---""22"",
+        // >--->---33,
+        // >--->---[]
+        // >---],
+        // >---4
+        // ]
+    (end)
+
+Author:
+   Terra
+---------------------------------------------------------------------------- */
+SCRIPT(prettyFormat);
+params [["_array", [], [[]]], ["_tab", toString [9], [""]], ["_depth", 0, [0]]];
+private _tabs = [];
+_tabs resize _depth;
+_tabs = _tabs apply {_tab} joinString "";
+if (_array isEqualTo []) exitWith {
+    _tabs + "[]"
+};
+private _lines = [_tabs + "["];
+private _lastIndex = count _array - 1;
+{
+    private _line = "";
+    if (_x isEqualType []) then {
+        _line = _line + ([_x, _tab, _depth + 1] call CBA_fnc_prettyFormat);
+    } else {
+        _line = _tab + _tabs + _line + str _x;
+    };
```

Comment:
```suggestion
    private "_line";
    if (_x isEqualType []) then {
        _line = [_x, _tab, _depth + 1] call CBA_fnc_prettyFormat;
    } else {
        _line = _tab + _tabs + str _x;
    };
```

Reply:
Wouldn't this be a use case for using the return of the if statement as done in the original code?
```sqf
private _line = if (...
```

Reply:
this is exactly why I asked commy above. I would run some tests on large arrays with subarrays (e.g. `configClasses` can return huge arrays) for both your and commy's cases and if the difference is not so big I prefer commy's one just for readability. If tests show significant difference your case is preferred.

Reply:
From my tests there is an improvement by using the `_var = if ` version. Tested with this code:
```sqf
_array = getUnitLoadout (configFile >> "CfgVehicles" >> "B_Soldier_A_F");
diag_codePerformance [CBA_fnc_prettyFormat, [_array]];
```